src/parser/core/Combatant.ts

```diff
@@ -74,7 +74,12 @@ class Combatant extends Entity {
     if (!this.owner.characterProfile) {
```

Comment:
```suggestion
    if (!this.owner.characterProfile || ! this.owner.characterProfile.race) {
```

In that case I would recommend doing this instead. We already have to consider the race being unknown as we can't always query the character from the Armory. Things would be simpler if there was only 1 such situation for this in our code.